xcell/mappers/mapper_WIxSC.py

```diff
@@ -0,0 +1,248 @@
+                                          zflag='Zspec',
+                                          zrange=[0, 1.],
+                                          nz=100,
+                                          bands_photo=['W1MCORR', 'W2MCORR',
+                                                       'BCALCORR', 'RCALCORR'],
+                                          njk=self.config.get('n_jk_dir', 100))
+                zm = 0.5*(z[1:] + z[:-1])
+                if f_name is not None:
+                    np.savez(f_name, z_mid=zm, nz=nz, nz_jk=nz_jk)
+            self.dndz = (zm, nz, nz_jk)
+
+        z, nz, nz_jk = self.dndz
+        z_dz = z + dz
+        sel = z_dz >= 0
+        if return_jk_error:
+            njk = len(nz_jk)
+            enz = np.std(nz_jk, axis=0)*np.sqrt((njk-1)**2/njk)
+            return np.array([z_dz[sel], nz[sel], enz[sel]])
+        else:
+            return np.array([z_dz[sel], nz[sel]])
+
+    def get_signal_map(self):
+        if self.delta_map is None:
+            d = np.zeros(self.npix)
+            self.cat_data = self.get_catalog()
+            self.mask = self.get_mask()
+            self.stars = self._get_stars()
+            nmap_data = get_map_from_points(self.cat_data, self.nside,
+                                            ra_name='RA', dec_name='DEC',
+                                            in_radians=True)
+            mean_n = self._get_mean_n(nmap_data)
+            goodpix = self.mask > 0
+            # Division by mask not really necessary, since it's binary.
+            d[goodpix] = nmap_data[goodpix]/(mean_n*self.mask[goodpix])-1
+            if self.config.get('apply_galactic_correction', True):
+                gcorr = self._get_galactic_correction(d, self.stars,
+                                                      self.mask)
+                d -= gcorr['delta_map']
+            self.delta_map = d
+        return [self.delta_map]
+
+    def get_mask(self):
+        if self.mask is None:
+            self.mask = hp.ud_grade(hp.read_map(self.config['mask'],
+                                                verbose=False),
+                                    nside_out=self.nside)
+        return self.mask
+
+    def _get_stars(self):
+        if self.stars is None:
+            # Power = -2 makes sure the total number of stars is conserved
+            self.stars = hp.ud_grade(hp.read_map(self.config['star_map'],
+                                                 verbose=False),
+                                     nside_out=self.nside, power=-2)
+            # Convert to stars per deg^2
+            pix_srad = 4*np.pi/self.npix
+            pix_deg2 = pix_srad*(180/np.pi)**2
+            self.stars /= pix_deg2
+        return self.stars
+
+    def _get_mean_n(self, nmap):
+        self.stars = self._get_stars()
+        self.mask = self.get_mask()
+        goodpix = self.stars < 8515
+        goodpix *= self.mask > 0
+        n_mean = np.sum(nmap[goodpix])/np.sum(self.mask[goodpix])
```

Comment:
np.average() maybe?

Reply:
humm, this shouldn't apply here. Here we're not computing `sum(n*m)/sum(m)`, but `sum(n)/sum(m)`. The point is that in this case `nmap` is already multiplied by the mask, which is not binary (i.e. it incorporates any variations in the local number of galaxies).